### PR TITLE
Allow non-ascii keys

### LINKED ISSUE #118: aiomcache not support chinese bytes as a key
I want use chinese bytes as a key:

```python
In [1]: import asyncio 
   ...: import aiomcache 
   ...:  
   ...: loop = asyncio.get_event_loop()                                                                                                                                                                                                                                             

In [2]: async def hello_aiomcache(): 
   ...:     mc = aiomcache.Client('127.0.0.1', 11211, loop=loop) 
   ...:     await mc.set(bytes('中文', 'utf-8'), b'123') 
   ...:                                                                                                            

In [3]: loop.run_until_complete(hello_aiomcache())                                                                 
---------------------------------------------------------------------------
ValidationException                       Traceback (most recent call last)
<ipython-input-5-adb51212a606> in <module>
----> 1 loop.run_until_complete(hello_aiomcache())

/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py in run_until_complete(self, future)
    465             raise RuntimeError('Event loop stopped before Future completed.')
    466 
--> 467         return future.result()
    468 
    469     def stop(self):

<ipython-input-4-d36e59b6000e> in hello_aiomcache()
      1 async def hello_aiomcache():
      2     mc = aiomcache.Client('127.0.0.1', 11211, loop=loop)
----> 3     await mc.set(bytes('中文', 'utf-8'), b'123')
      4 

~/.virtualenvs/lyanna-tp9VnnDu/lib/python3.6/site-packages/aiomcache/client.py in wrapper(self, *args, **kwargs)
     18         conn = yield from self._pool.acquire()
     19         try:
---> 20             return (yield from func(self, conn, *args, **kwargs))
     21         except Exception as exc:
     22             conn[0].set_exception(exc)

~/.virtualenvs/lyanna-tp9VnnDu/lib/python3.6/site-packages/aiomcache/client.py in set(self, conn, key, value, exptime)
    250         flags = 0  # TODO: fix when exception removed
    251         resp = yield from self._storage_command(
--> 252             conn, b'set', key, value, flags, exptime)
    253         return resp
    254 

~/.virtualenvs/lyanna-tp9VnnDu/lib/python3.6/site-packages/aiomcache/client.py in _storage_command(self, conn, command, key, value, flags, exptime, cas)
    218         # however custom-compiled memcached can have different limit
    219         # so, we'll let the server decide what's too much
--> 220         assert self._validate_key(key)
    221 
    222         if not isinstance(exptime, int):

~/.virtualenvs/lyanna-tp9VnnDu/lib/python3.6/site-packages/aiomcache/client.py in _validate_key(self, key)
     54                 raise ValidationException('trailing newline', key)
     55         else:
---> 56             raise ValidationException('invalid key', key)
     57 
     58         return key

ValidationException: invalid key: b'\xe4\xb8\xad\xe6\x96\x87'

In [4]: debug                                                                                                      
> /Users/dongweiming/.virtualenvs/lyanna-tp9VnnDu/lib/python3.6/site-packages/aiomcache/client.py(56)_validate_key()
     54                 raise ValidationException('trailing newline', key)
     55         else:
---> 56             raise ValidationException('invalid key', key)
     57 
     58         return key

ipdb> key                                                                                                          
b'\xe4\xb8\xad\xe6\x96\x87'
```

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Fixes #118.

### CODE REVIEW

**Comment by Dreamsorcerer** [comment]
@webknjaz LGTM checks blocking merging is a little annoying. We have flake8 setup now, so it should rarely trigger any errors anyway.

**Comment by webknjaz** [comment]
Doesn't it make LGTM even more significant? I mean that it catches things flake8 missed? It's not like WPS linter set is integrated.

File: tests/commands_test.py [hunk 1]
```diff
@@ -7,6 +7,42 @@
...
+    # Control characters
+    b"foo\x00bar",
+    b"\x1F",
+    b"\x7F",
+    "\u0080".encode(),
+    "\u009F".encode(),
+))
+@pytest.mark.asyncio
+async def test_invalid_key(mcache, key):
+    with pytest.raises(ValidationException):
```
Comment (webknjaz) [inline tests/commands_test.py:42]: It's best to always use the `match` arg in `pytest.raises()` and `pytest.warns()`.

**Comment by Dreamsorcerer** [comment]
> Doesn't it make LGTM even more significant? I mean that it catches things flake8 missed? It's not like WPS linter set is integrated.

Problem is that it takes 15+ mins to complete (compared with <1 min for everything else), and will find an issue in probably about 1 in 500 PRs. I'd rather merge smaller PRs without waiting, and in the unlikely chance of it finding an issue, it can be fixed in a followup PR. I think this is how all the other repos are being used as well, including aiohttp.

**Comment by webknjaz** [comment]
@Dreamsorcerer more complex checks usually need more resources so it's kinda expected. I wouldn't judge the quality/usefulness of a check by this metric. Moreover, staring at the CI log in order to catch it as soon as it turns green, is a waste of time for a human — you shouldn't need to babysit it, it's done when it's done. If the CI is blocking you from performing other unrelated activities, you're likely doing something wrong.

Instead, let automation do the merge if the checks pass, it doesn't really require any attention from you. Use the `Auto-merge` feature that is enabled in the repository once there's a human approval and then, just walk away. Hopefully, when GH is done alpha-testing Zuul-like merge queues, it'll become way easier to deal with batches of PRs the same way Zuul does this.

FWIW the human's job is to perform a human review and the checks are on the automation (https://mtlynch.io/human-code-reviews-1/#let-computers-do-the-boring-parts) — you look into them only if and after something breaks.

**Pull request merged**